### PR TITLE
Preliminary SVR Prediction Feature

### PR DESCRIPTION
**Pull request by lussierc**
This PR implements the preliminary version of the stock price swing prediction feature that was discussed extensively in #6. The tool can now generate a stock price prediction and price swing prediction using a SVR that is trained using previous date and closing price data that is scraped from Yahoo! Finance.

The stock price swing prediction model is the main feature of the tool, helping both new and experienced stock traders make more informed trading decisions. The stock price swing prediction rendered by the program will give traders insights into how a stock price is going to move, allowing them to make better buying and selling decisions while having a higher chance of making a profit.

*Features/changes implemented in this PR include:*
- Create `prediction.py` and `data_cleaner.py` files to hold prediction and data cleaning code.
- Implement code needed to generate a stock price prediction and stock price swing prediction for the next day in the `prediction.py` file.
   - Use a Support Vector Regression (SVR) to generate these predictions, which is implemented using the `sklearn.svm` library.
      - Tries three types of SVRs, including `rbf`, `linear`, and `poly` kernels. With this, the `rbf` kernel was identified as by far the most accurate of the three and will be used to generate the stock price swing prediction.
         - The results of using these models to predict previous day's stock prices are plotted using the `matplotlib.pyplot` library. 
   - When the `run_predictor()` function is given data scraped by the tool, it will run all the other functions necessary to generate a prediction.
- Improved the way scraped data is stored by storing each individual stock's data within it's own dictionary. This dictionary contains the stock name, current price, historical data, and more.
- Implemented functions in `data_cleaner.py` that prepare data for use with the prediction model.
- Created a resources folder to hold project images/resources. Also update README with a basic graph example.

### CODE REVIEW

**Comment by lussierc** [comment]
Here is an image of a graph generated by the project:

Graph

In this graph example, the SVR models run through the previous days data and work to predict their prices. Finally, a prediction for the next day's price is generated. With this, clearly, the model using the `rbf` seems to be the most accurate. The stock in this graph is Draftkings (DKNG) and is using the last 60 days worth of data.

**Comment by lussierc** [comment]
As mentioned in Issue #6, here is some possible work that could be done to improve the stock price swing prediction and it's accuracy:

> So the `sklearn` library also offers an entire `sklearn.svm` sub-lib. This library offers a number of different types of SVMs that can be easily implemented, which are detailed [here](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.svm). Other or more SVM options could be considered from this library.
>
>With this, one thing I have been considering is combining SVM/ML methods to generate an accurate prediction after I have a basic SVR prediction model implemented. This could involve combining the results from multiple SVMs from the aforementioned `sklearn` library to generate a more accurate prediction or combining these SVMs with things like Neural Networks (NNs). Neural Networks were another prediction option I identified through my research that seemed to provide strong results. I like the idea of combing multiple ML methods as it would likely create a tool that is more applicable to more scenarios and a tool that is more accurate. In having a multi-layer prediction model, like that of an Artificial Neural Network, I could potential create a stronger tool.

One other area of future work I will definitely need to complete is in the area of testing the prediction feature and it's associated code files. This needs to be done ASAP, once the prediction tool is finalized and no major structural changes are anticipated to be made to the code. Testing is discussed more in #9.
